File: Тренировка 1.0/02. Линейный поиск/C. Ближайшее число/Ближайшее число.cpp
#include <iostream>
#include <cmath>
#include <vector>

using namespace std;

int main()
{
    short N;
    vector<short> list;
    cin >> N;
    while (N > 0)
    {
        short number;
        cin >> number;
        list.push_back(number);
        N--;
    }

    short x, closest = list[0];
    cin >> x;
    for (int i = 1, n = list.size(); i < n; i++)
    {
        if (list[i] == x)
        {
            closest = list[i];
            break;
        }
        else if (abs(closest - x) > abs(list[i] - x)) closest = list[i];
    }

    cout << closest;
}

File: Тренировка 1.0/02. Линейный поиск/C. Ближайшее число/Ближайшее число.cs
using System;

short N = short.Parse(Console.ReadLine());
string[] numbers = Console.ReadLine().Split(' ');
short x = short.Parse(Console.ReadLine());

short closest = short.Parse(numbers[0]);
for(int i = 1; i < N; i++)
{
    short number = short.Parse(numbers[i]);
    if (number == x)
    {
        closest = number;
        break;
    }
    else if (Math.Abs(closest - x) > Math.Abs(number - x)) closest = number;
}

Console.WriteLine(closest);